AWSIoT/AWSIoTDataManager.m

```diff
@@ -333,6 +333,28 @@ - (void)enableMetricsCollection:(BOOL)enabled {
     [self.mqttClient setIsMetricsEnabled:enabled];
 }
 
+- (void)addUserMetaData:(NSDictionary<NSString *, NSString *> *)userMetaData {
+
+    // validate the length of username field
+    NSMutableString *userMetadata = [NSMutableString stringWithFormat:@"%@%@", @"?SDK=iOS&Version=", [self.mqttClient SDK_VERSION]];
+
+    // Append each of the user-specified key-value pair to the connection username
+    if (userMetaData != [ NSNull null ]) {
+        for (id key in userMetaData) {
+            [userMetadata appendFormat:@"&%@=%@", key, [userMetaData objectForKey:key]];
+        }
+    }
+    NSString *baseUsername = [NSString stringWithFormat:@"%@%@", @"?SDK=iOS&Version=", [self.mqttClient SDK_VERSION]];
```

Comment:
Remove this code duplication. You can capture the length in a local variable after line 339 and use that to drive the logic.